logits_processor_zoo/transformers/generation_length.py

```diff
@@ -38,17 +39,20 @@ class GenLengthLogitsProcessor:
     """
     def __init__(self, tokenizer: PreTrainedTokenizer, boost_factor: float,
                  p: int = 2, complete_sentences: bool = False, boost_token_str: str = None):
+        super().__init__()
         self.boost_token = tokenizer.eos_token_id
         if boost_token_str is not None:
             self.boost_token = text_to_token(tokenizer, boost_token_str, last=False)
         self.boost_factor = boost_factor
         self.p = p
-        self.token_count = 0
         self.full_stop_token = text_to_token(tokenizer, "It is a sentence.", last=True)
         self.new_line_token = text_to_token(tokenizer, "It is a new line\n", last=True)
         self.complete_sentences = complete_sentences
 
-    def __call__(self, input_ids: torch.LongTensor, scores: torch.FloatTensor) -> torch.Tensor:
+    def _reset(self):
+        self.token_count = 0
```

Comment:
I'd either define `self.token_count = 0` in `init` (same for other processors) or call `self._reset() ` in init to achieve the same. Setting the attributes explicitly is probably the easiest solution (as one wouldn't need to worry/keep in mind to call _reset).


Reason: In some cases (e.g. tests), `._process` might be called instead of `__call__` which would then yield to an error.

Reply:
Since they import the base class, _reset is guaranteed to be called on behalf of the user.

We actually dont want users to call _process, therefore I named it with underscore to make it a private function.